frontend/src/metabase/common/components/EntityPicker/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,67 @@
+import { useLayoutEffect, useState } from "react";
+import { useDebounce } from "react-use";
+
+import { useSearchQuery } from "metabase/api";
+import { Icon, TextInput } from "metabase/ui";
+import type {
+  SearchModel,
+  SearchRequest,
+  SearchResult,
+} from "metabase-types/api";
+
+interface Props {
+  searchQuery: string;
+  setSearchQuery: (query: string) => void;
+  setSearchResults: (results: SearchResult[] | null) => void;
+  models: SearchModel[];
+  placeholder: string;
+  searchFilter?: (results: SearchResult[]) => SearchResult[];
+  searchParams?: Partial<SearchRequest>;
+}
+
+const defaultSearchFilter = (results: SearchResult[]) => results;
+
+export function SearchInput({
```

Comment:
This component has been moved from `EntityPickerSearch.tsx`.
What's been changed:
- Removed `EntityPicker` prefix
- `Props` are no longer defined inline

The rest is the same.